File: emu-manager-suite/frontend/src/lib/skillSp.ts
/** EVE Online cumulative skill points at a given trained level. */
export function cumulativeSkillSp(level: number, rank: number): number {
  if (level <= 0 || rank <= 0) return 0;
  let total = 0;
  for (let n = 1; n <= level; n += 1) {
    if (n === 1) {
      total += 250 * rank;
    } else {
      total += 250 * rank * 2 ** ((2 * n - 3) / 2);
    }
  }
  return Math.round(total);
}

export function maxSkillSp(rank: number): number {
  return cumulativeSkillSp(5, rank);
}

export function fmtSp(value: number): string {
  return value.toLocaleString(undefined, { maximumFractionDigits: 0 });
}

File: emu-manager-suite/backend/app/services/skill_sp.py
"""Skill point calculations."""

from __future__ import annotations


def cumulative_skill_sp(level: int, rank: int) -> int:
    if level <= 0 or rank <= 0:
        return 0
    total = 0.0
    for n in range(1, level + 1):
        if n == 1:
            total += 250 * rank
        else:
            total += 250 * rank * (2 ** ((2 * n - 3) / 2))
    return round(total)


def max_skill_sp(rank: int) -> int:
    return cumulative_skill_sp(5, rank)
